krummstab/config.py

```diff
@@ -1,71 +1,63 @@
 import json
 import logging
+from importlib import resources
 from pathlib import Path
-import re
-import types
-from typing import Union, Callable, get_origin, get_args
+import jsonschema
+from . import schemas
 
 Student = tuple[str, str, str]
 Team = list[Student]
 
 
-def ensure_list(value):
-    if not value:
-        return []
-    elif isinstance(value, list):
-        return value
-    else:
-        return [value]
-
-
 class Config:
     def __init__(self, config_paths: list[Path]) -> None:
+        schema_config_shared = json.loads(
+            resources.read_text(schemas, "config-shared-schema.json", encoding="utf-8"))
+        schema_config_individual = json.loads(
+            resources.read_text(schemas, "config-individual-schema.json", encoding="utf-8"))
+        jsonschema.validate(json.loads(config_paths[0].read_text(encoding="utf-8")),
+                            schema_config_shared, jsonschema.Draft7Validator)
+        jsonschema.validate(json.loads(config_paths[1].read_text(encoding="utf-8")),
```

Comment:
Die Funktion sollte mit einer beliebigen Menge von `config_paths` funktionieren. Die Idee ist, dass man in alle files alle Optionen schreiben kann. Sie werden aber der Reihe nach eingelesen, also hat ein Eintrag in der individuellen config eine höhere Priorität und überschreibt frühere Einträge. Eine Möglichkeit wäre zuerst alles zu lesen (Zeilen 22-26) und danach das kombinierte Ergebnis davon zu validieren. Es sollte dann nur ein Schema geben.